src/ctapipe/image/toymodel.py

```diff
@@ -375,20 +375,83 @@ def pdf(self, x, y):
 class RingGaussian(ImageModel):
     """A shower image consisting of a ring with gaussian radial profile.
 
-    Simplified model for a muon ring.
+    Simplified model for a muon ring. With asymmetry in both the x and y directions.
     """
 
-    def __init__(self, x, y, radius, sigma):
+    def __init__(
+        self,
+        x,
+        y,
+        radius,
+        sigma,
+        asymmetry_magnitude=0,
+        asymmetry_orientation_angle_deg=0,
+    ):
         self.unit = x.unit
         self.x = x
         self.y = y
         self.sigma = sigma
         self.radius = radius
+        self.asymmetry_slope_x = asymmetry_magnitude * np.cos(
+            asymmetry_orientation_angle_deg.to_value(u.rad)
+        )
+        self.asymmetry_slope_y = asymmetry_magnitude * np.sin(
+            asymmetry_orientation_angle_deg.to_value(u.rad)
+        )
         self.dist = norm(
             self.radius.to_value(self.unit), self.sigma.to_value(self.unit)
         )
 
     def pdf(self, x, y):
         """2d probability for photon electrons in the camera plane."""
         r = np.sqrt((x - self.x) ** 2 + (y - self.y) ** 2)
-        return self.dist.pdf(r)
+        uniform_ring_pdf = self.dist.pdf(r)
+        asymmetric_ring_pdf = (
+            uniform_ring_pdf
+            * self.linear_pdf(x, self.asymmetry_slope_x)
+            * self.linear_pdf(y, self.asymmetry_slope_y)
+        )
+        return (
+            asymmetric_ring_pdf / np.sum(asymmetric_ring_pdf) * np.sum(uniform_ring_pdf)
+        )
+
+    def linear_pdf(self, var, linear_slope=0):
+        """This is a function that generates a probability density
+        function (pdf) with a linear increase/decrease and a given slope in 1D.
+
+        Parameters
+        ----------
+        var : u.Quantity[length, shape(n, )]
+              A random variable with a linearly increasing or decreasing probability
+              density function.
+        linear_slope: linear slope
+
+        Returns
+        -------
+        a pdf shape(n, )
+
+        """
+
+        if linear_slope == 0:
+            return 1
+
+        if len(var) < 2:
+            raise ValueError("The var array is too short.")
+
+        min_var = np.nanmin(var)
+        max_var = np.nanmax(var)
+
+        if np.isnan(var).any():
```

Comment:
move this check above the min/max computation and use np.min and np.max instead of nanmin and nanmax

Reply:
not after the check. Please change the nanmin and nanmax to np.min and np.max, they are about twice faster.